src/snyk/cli/services/cliService.ts

```diff
@@ -39,18 +40,27 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
       return error;
     }
 
-    const process = new CliProcess(this.logger, this.config);
+    if (this.cliProcess) {
+      const killed = this.cliProcess.kill();
+      if (!killed) this.logger.error('Failed to kill an already running CLI instance.');
```

Comment:
shall we add a timer interval here to try and kill again after X ms?

Reply:
also, if we failed killing an already running CLI instance, what would be the outcome in this case?

Reply:
Simply trying again would not help killing the process. Kill doesn't succeed usually if a process was already terminated or some other error occurred (e.g. EPERM, or underlying platform doesn't support killing the process).

If we failed at killing an already running CLI instance, the process will bring the analysis to the end, terminating the process when finished.